### PR TITLE
[DOC] Transform documentation to .md

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue
Documentation needs to be converted to .md format due to migration of the website to Hugo.

##### Description of changes
- [X] Data section to .md
- [x] Visualize section to .md
- [x] Model section to .md
- [x] Evaluate section to .md
- [x] Unsupervised section to .md

##### Includes
- [ ] Code changes
- [ ] Tests
- [X] Documentation

### CODE REVIEW

**Comment by BlazZupan** [comment]
This looks nice: the structure of the documentation is simplified, the writing is easier to read than RST. Time to finish and close.

**Comment by ajdapretnar** [comment]
Tests fail because there is no recommonmark in the requirements. However, we removed it 3 months ago. https://github.com/biolab/orange3/commit/97b1a750a969443ad134d56e88f2c4e6be979ed6
Perhaps @ales-erjavec you can hint at what I should do? Put it back in the requirements? From what I remember it was breaking things. 🤔

**Comment by ales-erjavec** [comment]
try
```diff
diff --git a/appveyor.yml b/appveyor.yml
index 894648cfe..d7eb3c9cf 100644
--- a/appveyor.yml
+++ b/appveyor.yml
@@ -18,7 +18,7 @@ environment:
     # disable threaded builds on Python 3.5+ when using numpy's distutils
     # (https://github.com/numpy/numpy/issues/7607)
     BUILD_GLOBAL_OPTIONS: build -j1
-    BUILD_ENV: wheel==0.29.0 pip==9.0.1 numpy==1.9.3 sphinx==1.8.2
+    BUILD_ENV: wheel==0.29.0 pip==9.0.1 numpy==1.9.3 -r requirements-doc.txt
     # SIP 4.19.4+ with PyQt5==5.9.1+ segfault our tests (GH-2756)
     TEST_ENV: sip==4.19.6 PyQt5==5.9.2 numpy~=1.14.0 scipy~=1.0.0 scikit-learn pandas==0.21.1
 
diff --git a/requirements-dev.txt b/requirements-dev.txt
index b5637ead3..2ea73d2f7 100644
--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -1,4 +1,5 @@
 pylint
 radon
 # for build_htmlhelp command
-sphinx>=1.5
\ No newline at end of file
+sphinx>=1.5
+recommonmark
diff --git a/requirements-doc.txt b/requirements-doc.txt
index dcc31f22c..047b80f47 100644
--- a/requirements-doc.txt
+++ b/requirements-doc.txt
@@ -1,4 +1,5 @@
 docutils
 numpydoc
 Sphinx>=1.3
+recommonmark
 PyQt5
diff --git a/setup.py b/setup.py
index 76cf0be7b..50a4a6f7a 100755
--- a/setup.py
+++ b/setup.py
@@ -17,7 +17,9 @@ except ImportError:
 
 
 try:
+    # need sphinx and recommonmark for build_htmlhelp command
     from sphinx.setup_command import BuildDoc
+    import recommonmark
     have_sphinx = True
 except ImportError:
     have_sphinx = False

```

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 pylint
 radon
 # for build_htmlhelp command
-sphinx>=1.5
\ No newline at end of file
+sphinx>=1.5
+recommonmark
```
Comment (ales-erjavec) [inline requirements-dev.txt]: `requirements-dev.txt` requires a newline at the end (for the benefit of our travis ci scripts).

**Comment by ajdapretnar** [comment]
@ales-erjavec This seems like it is failing due to some unrelated errors + the recommonmark import we added. Is there a nice way to ignore it or fix it otherwise? I have additionally fixed conf.py to address the pylint complaints, but I didn't dare to touch setup.py.

**Comment by ales-erjavec** [comment]
```diff
diff --git a/setup.py b/setup.py
index 50a4a6f7a..8a97e9c0e 100755
--- a/setup.py
+++ b/setup.py
@@ -19,6 +19,7 @@ except ImportError:
 try:
     # need sphinx and recommonmark for build_htmlhelp command
     from sphinx.setup_command import BuildDoc
+    # pylint: disable=unused-import
     import recommonmark
     have_sphinx = True
 except ImportError:
```

**Comment by ajdapretnar** [comment]
Thank you for noticing! I thought Sphinx would check this, but the build passed without complaints, when I ran it locally.
It seems like everything passes now. I will fix the link and push once again. Then someone can accept this PR, I guess.

**Comment by ales-erjavec** [comment]
It does warn about it: https://travis-ci.org/biolab/orange3/jobs/480437250#L2425

There are other warnings regarding links to images which are seem to be a result of different capitalization (https://travis-ci.org/biolab/orange3/jobs/481266680#L1981).

**Comment by markotoplak** [comment]
@ajdapretnar, you probably do not have a case sensitive file system on your computer, so you might not see the errors locally.

**Comment by ales-erjavec** [comment]
The other warnings about the .tab files are due to recommonmark not processing the link destination as a resource to be moved to the build directory. I would suggest moving the *.tab files to `_static` and updating the links to `../../_static/*.tab`.

**Comment by ajdapretnar** [comment]
It is true what @markotoplak said - my system does not complain regarding the links. I am using VS Code to visualize and test the links and they all worked even before the transformation.

I have moved data to _static.
I also transformed some new .rst files to .md while I was at it.
However, I don't know how to fix the images. Well, I do, but the first option (which I prefer) would be to rename the images. Once I do, my system doesn't recognize the difference between the old and new file, so I cannot make a commit. Of course, I could fix the naming in the files, but I would rather streamline the naming in the images, which is currently inconsistent. If this is too much hassle, I can of course just rename the links and be done with this.

**Comment by ales-erjavec** [comment]
Changing just the capitalization of a filename usually result in pain for git on case insensitive file systems. So it would be best to change the links.

**Pull request merged**